blueocean-dashboard/src/main/js/util/UrlUtils.js

```diff
@@ -173,11 +173,12 @@ export function getRestUrl({ organization, pipeline, branch, runId }) {
     const jenkinsUrl = require('../config').getJenkinsRootURL();
     let url = `${jenkinsUrl}/blue/rest/organizations/${encodeURIComponent(organizationName)}`;
     if (pipelineName) {
-        // pipelineName might include a folder path, don't escape it
+        // pipelineName might include a folder path, don't encode it
         url += `/pipelines/${pipelineName}`;
     }
     if (branch) {
-        url += `/branches/${encodeURIComponent(branch)}`;
+        // JENKINS-37712 branch needs to be double-encoded for some reason
+        url += `/branches/${encodeURIComponent(encodeURIComponent(branch))}`;
```

Comment:
@tfennelly it has been in there since the start, stapler requires double encoding (!) in paths (the URIs in blue ocean do not, which in retrospect seems a mistake to be different) - this was talked about in another PR. @cliffmeyers I believe has some WIP to centralise the URL calculation so this doesn't have to happen.